ecl/hql/hqlgram2.cpp

```diff
@@ -12311,8 +12311,9 @@ void parseAttribute(IHqlScope * scope, IFileContents * contents, HqlLookupContex
         throw;
     }
     OwnedHqlExpr parsed = scope->lookupSymbol(name, LSFsharedOK|LSFnoreport, ctx);
-    bool canCache = parsed && (parsed->getOperator() != no_forwardscope) && !alreadySimplified;
-    bool isMacro = parsed && parsed->isMacro();
+    const auto no = parsed ? parsed->getOperator() : no_none;
+    const bool canCache = (no!=no_none) && !alreadySimplified && (no!=no_forwardscope) && (no!=no_virtualscope) ;
```

Comment:
I don't think no_virtualscope is correct in this list.
However if parsed is a function then this should be looking at the body to find out if that is a forward scope.